src/Element.js

```diff
@@ -88,7 +88,13 @@ Element.prototype.attributeNameMappings = {
 }
 
 Element.prototype.attributeToPropName = function (name) {
-  return this.attributeNameMappings[name] || camelCase(name)
+  if (name.match(/^data-/)) {
```

Comment:
I forgot to mention before I merged it that these RegExps should probably be initialised in a closure (cached and reused).